garak/generators/rest.py

```diff
@@ -256,18 +276,32 @@ def _call_model(self, prompt: str, generations_this_call: int = 1):
                 raise ConnectionError(error_msg)
 
         if not self.response_json:
-            return str(resp.content)
-
-        try:
-            response_object = json.loads(resp.content)
-            return response_object[self.response_json_field]
-        except json.decoder.JSONDecodeError as e:
-            logging.warning(
-                "REST endpoint didn't return good JSON %s: got |%s|",
-                str(e),
-                resp.content,
-            )
-            return None
+            return str(resp.text)
+
+        response_object = json.loads(resp.content)
+
+        # if response_json_field starts with a $, treat is as a JSONPath
+        if self.response_json_field[0] != "$":
```

Comment:
Might be worth guarding this for `None` or empty. Consider adding the guards in `__init__()` above to avoid possible user config error.